File: backend/app/api/dtos/data_api_response.py
from __future__ import annotations

from typing import Generic, TypeVar

from app.api.dtos.base_api_response import BaseApiResponseDto

T = TypeVar("T")


class DataApiResponseDto(BaseApiResponseDto, Generic[T]):
    """
    Data API response DTO.

    Extends BaseApiResponseDto to include a generic `data` payload.
    """

    data: T | None = None

File: frontend/src/types/dtos/base/data.api.response.ts
import { BaseApiResponseDto } from '@src/types/dtos/base/base.api.response';

/**
 * Data API response DTO.
 * Extends BaseApiResponseDto to include a generic `data` payload.
 */
export interface DataApiResponseDto<T> extends BaseApiResponseDto {
  data?: T | null;
}
